src/coreclr/vm/prestub.cpp

```diff
@@ -2746,12 +2746,16 @@ extern "C" void STDCALL ExecuteInterpretedMethod(TransitionBlock* pTransitionBlo
     InterpThreadContext *threadContext = InterpGetThreadContext();
     int8_t *sp = threadContext->pStackPointer;
 
-    InterpMethodContextFrame interpFrame = {0};
-    interpFrame.startIp = (int32_t*)byteCodeAddr;
-    interpFrame.pStack = sp;
-    interpFrame.pRetVal = sp;
+    InterpMethodContextFrame interpMethodContextFrame = {0};
+    interpMethodContextFrame.startIp = (int32_t*)byteCodeAddr;
+    interpMethodContextFrame.pStack = sp;
+    interpMethodContextFrame.pRetVal = sp;
 
-    InterpExecMethod(&interpFrame, threadContext);
+    InterpreterFrame interpreterFrame(pTransitionBlock, &interpMethodContextFrame);
+
+    InterpExecMethod(&interpreterFrame, &interpMethodContextFrame, threadContext);
```

Comment:
This change looks ok for now. Do we have a plan for where and how the arguments are going to converted from the `TransitionBlock` to the interpreter representation?

Reply:
I'm thinking when invoking an interpreted method through the precode thunks, instead of calling `ExecuteInterpretedMethod` like we do now, it will call some other assembly thunk that receives a lower level bytecode referenced from the interpreter bytecode (maybe in some Interpreter code header). These assembly thunks will move arguments from the transition block to the interpreter stack, finally dispatching to `ExecuteInterpreterMethod`. I described this mechanism a while ago in a doc. Last time I spoke with @janvorli, this plan was to use this mechanism from the start since we will need it as a fallback anyway on iOS.

Reply:
I've used the TransitionBlock here for two reasons. One, as a possibly easy starting point for passing arguments until we implement the proper mechanism and second, as a mean to save and restore the callee saved registers. This is necessary as we need to be able to restore them during stack walking from interpreted to JITted / AOTed methods. Once we have the optimized mechanism, we can use a smaller structure than the TransitionBlock that would contain only the callee saved registers.